### PR TITLE
[17.0] [ADD] account_vendor_bank_account_default_purchase

### PR DESCRIPTION
**Pull request by Tisho99**
New module: account_vendor_bank_account_default_purchase

The module allows users to set a default bank account purchase orders. 

Depends of #1292

Related with #1294 and #1326

T-6102

### CODE REVIEW

File: account_vendor_bank_account_default_purchase/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2024 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Account Vendor Bank Account Default Purchase",
+    "summary": "Set a default bank account purchase orders",
+    "version": "17.0.1.0.0",
+    "development_status": "Beta",
+    "category": "Banking addons",
+    "website": "https://github.com/OCA/bank-payment",
+    "author": "Alberto Martínez, Odoo Community Association (OCA)",
```
Comment (ValentinVinagre) [inline account_vendor_bank_account_default_purchase/__manifest__.py]: change by sygel please 👍🏻

**Review by ValentinVinagre — CHANGES_REQUESTED** [review]

File: account_vendor_bank_account_default_purchase/__manifest__.py (hunk 1 above)
Comment (Tisho99) [reply account_vendor_bank_account_default_purchase/__manifest__.py]: Done

File: account_vendor_bank_account_default_purchase/README.rst [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+1. Create a purchase order. On the other info tab you can see its bank
+   account, that will be equal to the configured in the partner's
+   Default Banking Account field
+2. Confirm the purchase and create its vendor bill. Its reccipient bank
+   account will be equal to the purchase's bank account
+
+Known issues / Roadmap
+======================
+
+-  Incompativility with account_payment_partner module
```
Comment (Jesarregui) [inline account_vendor_bank_account_default_purchase/README.rst]: Could you check this line?
I do not detect any incompatibility with this module.

**Comment by Tisho99** [comment]
Moving features of this PR to https://github.com/OCA/bank-payment/pull/1326

**Comment by Tisho99** [comment]
@HaraldPanten @manuelregidor 

The PR is ready

**Review by Jesarregui — APPROVED** [review]
Functional Review: LGTM

File: account_vendor_bank_account_default_purchase/models/purchase_order.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    def _get_default_supplier_partner_bank(self, partner):
+        if not partner.has_default_bank_id:
+            return False
+        elif partner.default_bank_id:
+            return partner.default_bank_id.id
+        else:
+            return super()._get_default_supplier_partner_bank(partner)
+
+    @api.depends("partner_id", "company_id")
```
Comment (manuelregidor) [inline account_vendor_bank_account_default_purchase/models/purchase_order.py:19]: I don't think you need to include the depends decorator when no new dependencies are added.

Comment (Tisho99) [reply account_vendor_bank_account_default_purchase/models/purchase_order.py:19]: I include it for readability, but i can remove it

Comment (manuelregidor) [reply account_vendor_bank_account_default_purchase/models/purchase_order.py:19]: Up to you really

**Review by manuelregidor — APPROVED** [review]
Technical review. LGTM

File: account_vendor_bank_account_default_purchase/readme/DESCRIPTION.md [hunk 4]
```diff
@@ -0,0 +1,2 @@
+This module allows you to set a default bank account purchase orders. 
+That bank account will be propagated to its vendor bills
```
Comment (HaraldPanten) [inline account_vendor_bank_account_default_purchase/readme/DESCRIPTION.md]:
```suggestion
That bank account will be propagated to its vendor bills.
```

File: account_vendor_bank_account_default_purchase/readme/USAGE.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+To use this module, you need to:
+
+1. Create a purchase order. On the other info tab you can see its bank account, which will be equal to the configured in the partner's "Default Bank Account" field
```
Comment (HaraldPanten) [inline account_vendor_bank_account_default_purchase/readme/USAGE.md]:
```suggestion
1. Create a purchase order. On the other info tab you can see its bank account, which will be equal to the configured in the partner's "Default Bank Account" field.
```

File: account_vendor_bank_account_default_purchase/readme/CONFIGURE.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+To configure this module, you need to:
+
+-  Go to a partner form view, and edit the Default Bank Account in the
+   Sales and Purchase tab. The partner must be of type 'company', or an
+   individual without a parent company.
```
Comment (HaraldPanten) [inline account_vendor_bank_account_default_purchase/readme/CONFIGURE.md]:
```suggestion
-  Go to a partner form view, and edit the Default Bank Account in the
   Sales and Purchase tab. The partner must be set as a 'company', or as an
   individual without a parent company. 
```

File: account_vendor_bank_account_default_purchase/readme/USAGE.md [hunk 7]
```diff
@@ -0,0 +1,4 @@
...
+2. Confirm the purchase and create its vendor bill. Its recipient bank account will be equal to the purchase's bank account
```
Comment (HaraldPanten) [inline account_vendor_bank_account_default_purchase/readme/USAGE.md]:
```suggestion
2. Confirm the purchase and create its vendor bill. Its recipient bank account will be equal to the purchase's bank account.
```

**Review by HaraldPanten — COMMENTED** [review]
Some picky comments

**Review by HaraldPanten — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**